### PR TITLE
Config option for ValidateFieldsOutOfOrder

### LINKED ISSUE #107: ValidateFieldsOutOfOrder Configuration Option
Necessary for counter parties that send FIX messages out of order

### PR DESCRIPTION
**Pull request by bhaan**
closes #107 

Changes proposed in this pull request:
- Add `ValidateFieldsOutOfOrder` to configuration
- Add `validatorSettings` to message validation
- Update `session` to import the `ValidateFieldsOutOfOrder` configuration flag
- Update `validateWalk` to no longer assume the message is in the correct order
- [x] Unit test for disabling `ValidateFieldsOutOfOrder`

### CODE REVIEW

File: config/doc.go [hunk 1]
```diff
@@ -130,5 +130,9 @@ The name of the database driver to use (see https://github.com/golang/go/wiki/SQ
 SQLDataSourceName
 
 The driver-specific data source name of the database to use.  Only used with SqlStoreFactory.
+
+ValidateFieldsOutOfOrder
+
+If set to N, fields that are out of order (i.e. body fields in the header, or header fields in the body) will not be rejected. Useful for connecting to systems which do not properly order fields.
```
Comment (cbusbey) [inline config/doc.go]: defaults to Y?

File: validation_test.go [hunk 2]
```diff
@@ -336,7 +336,7 @@ func tcIncorrectDataFormatForValue() validateTest {
 
 func tcTagSpecifiedOutOfRequiredOrder_Header() validateTest {
```
Comment (cbusbey) [inline validation_test.go]: maybe dup this test and the _trailer test but with validation turned off?

File: config/doc.go (hunk 1 above)
Comment (bhaan) [inline config/doc.go]: Sorry about that. I just added "Valid values" and the "Defaults to Y": c10779b

File: validation_test.go (hunk 2 above)
Comment (bhaan) [inline validation_test.go]: 👍  I agree: 9199547

**Comment by cbusbey** [comment]
🍹 🐻

**Pull request merged**